cpp/src/arrow/type_traits.h

```diff
@@ -858,7 +858,7 @@ using enable_if_physical_floating_point =
 ///
 /// \param[in] type_id the type-id to check
 /// \return whether type-id is an integer type one
-static inline bool is_integer(Type::type type_id) {
+static constexpr inline bool is_integer(Type::type type_id) {
```

Comment:
Constexpr implies `inline` and should not need the `static` quailfier
```suggestion
constexpr bool is_integer(Type::type type_id) {
```